### PR TITLE
Release v1

### PR DESCRIPTION
**Pull request by graingert**
this will be the tag:

```
tag v1.0.0
Tagger: Thomas Grainger <tagrain@gmail.com>
Date:   Wed Oct 21 11:25:04 2020 +0100

Tagging 1.0.0

commit f5fc2207d8ecf376284af01c6396c10f4dd2ffc4 (tag: v1.0.0)
```

### CODE REVIEW

**Comment by exarkun** [comment]
is there a document explaining how a release is produced?

**Comment by graingert** [comment]
@exarkun https://github.com/twisted/twisted-iocpsupport/blob/default/.github/workflows/github-deploy.yml#L66

You push a tag and off it goes to pypi

**Comment by graingert** [comment]
Some optional stuff like changelog and commit message can be done manually or with zest.releaser: https://github.com/twisted/ldaptor/pull/192/files

**Comment by adiroiban** [comment]
Copied from ldaptor

I think that this should be in the CONTRIBUTIONG.rst file.

```
The PyPI package publishing is done automatically via GitHub actions when a new tag is pushed.

To trigger the release for new version, follow this process:

* Create a new branch with the name `release-VER` (ex release-1.1.0) No need to create a separate issue.
* On that branch update the version from `setup.cfg ` to the final version to be released.
* Update the  CHANGELOG.rst file with the final version and release date.
* Push the commit and record the commit SHA as this will be the one used to tag and do the release.
* Include the tagged commit to the PR description.
* On the same branch, update the `setup.cfg ` to a `.dev0` version and update CHANGELOG.rst to a new empty unreleased section.
* Push the latest commit and request a review for the PR.
* Once the PR is approved, you can merge the PR
* After the merge, push the release tag. GitHub action will take care of the release.

You can check the GitHub actions release process here https://github.com/twisted/twisted-iocpsupport/actions

Below is the template text for the unreleased NEWS section::

    1.0.1 (unreleased)
    -------------------


    Changes
    - Nothing changed yet.

```

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+* On the same branch, update the `setup.cfg ` to a `.dev0` version and update CHANGELOG.rst to a new empty unreleased section.
+* Push the latest commit and request a review for the PR.
+* Once the PR is approved, you can merge the PR
+* After the merge, push the release tag. GitHub action will take care of the release.
+
+You can check the GitHub actions release process here https://github.com/twisted/twisted-iocpsupport/actions
+
+Below is the template text for the unreleased NEWS section::
+
+    1.0.1 (unreleased)
```
Comment (adiroiban) [inline CONTRIBUTING.rst:23]:
```suggestion
    1.0.1 (unreleased)
    ------------------

    - Nothing changed.
```

Comment (graingert) [reply CONTRIBUTING.rst:23]:
```suggestion
    1.0.1 (unreleased)
    ------------------

    - Nothing changed yet.
```

**Review by adiroiban — COMMENTED** [review]
I will let @exarkun do the final approval.

Below is a list of thinks missing from the current repo. Some are minor and maybe can be done later, but  I think that we should have GitHub Issues created for then,

* For this repo, do we use Twisted Trac Tickets or GitHub Issues?
* Update CONTRIBUTION to define that semver is used
* Update README to define the comparability policy 
* Update README to state that this is MIT (it make it easier) --- the current https://github.com/twisted/twisted-iocpsupport/blob/default/LICENSE is not detected by GitHub as MIT ... but is visible in setup.cfg
* Maybe create a GitHub PR template in which we provide contributing guidelines

I have updated the GitHub description, URL and tags for this repo. See the right column "About" section at https://github.com/twisted/twisted-iocpsupport

If we will used Twisted Trac Tickets, I will disabled the Issues link for this repo.

I have disabled the Wiki page for this repo... as I think that any documentation should be in twisted/twisted docs

**Comment by adiroiban** [comment]
For https://pypi.org/project/twisted-iocpsupport/ maybe also add exarkun , rodrigc  and itamarst  to the list of maintainers ... to be closer to the maintainers for https://pypi.org/project/Twisted/

Also, I think that README should be extend to provide a bit more info... the current PyPi page looks like for a alpha stage project while this is production ready ... maybe also update classifiers to explain that this is PROD ready https://pypi.org/project/twisted-iocpsupport/

**Comment by graingert** [comment]
> * For this repo, do we use Twisted Trac Tickets or GitHub Issues?

Twisted trac
> * Update CONTRIBUTION to define that semver is used

The versioning scheme is *only* consumed by https://github.com/twisted/twisted it's entirely an implementation detail to make `pip install twisted` download the right thing.
> * Update README to define the comparability policy

I see the policy as: twisted-iocpsupport will *never* break compatibility with Twisted. But the entire package is private and an implementation detail of `twisted`. Applications that are not `twisted` must not depend directly on `twisted-iocpsupport` or import any of its members.

> * Update README to state that this is MIT (it make it easier) --- the current [`default`/LICENSE](https://github.com/twisted/twisted-iocpsupport/blob/default/LICENSE) is not detected by GitHub as MIT ... but is visible in setup.cfg

with permission I can just copy in the standard MIT licence (like https://github.com/twisted/ldaptor/blob/master/LICENSE) and reference "contributors" using a https://git.io/ to the twisted licence

> * Maybe create a GitHub PR template in which we provide contributing guidelines

Probably just a link to "https://github.com/twisted/twisted"

**Comment by adiroiban** [comment]
I wrote down all my observation here.

Feel free to release this with the minimum changes. I can then followup with a PR to update README / Pull request template ...etc

But maybe it would be nice to have the README updated now, as I don't know when we will make a new release.

For the README my suggestion:

```


twisted-iocpsupport
===============

PYPI BADGE

This code was initially part of the core Twisted project.

Issues are handled by Twisted Trac Ticketing system https://twistedmatrix.com/trac/search?q=iocp&noquickjump=1&ticket=on

It was moved into a separate repo in order to simplify the Twisted production deployment.

The code is used to implemented the Twisted IOCP reactor.

The code is generic enough and in theory it can be used outside of the Twisted projects.

HERE INFORM IF THIS IS A TWISTED private package or we accept contribution for outside Twisted
```

**Comment by graingert** [comment]
@adiroiban I rebased so the v1.0.0 release is after the README change

File: README.rst [hunk 2]
```diff
@@ -1 +1,25 @@
-An extension for use in the L{twisted.internet.iocpreactor} I/O Completion Ports reactor.
+twisted-iocpsupport
+===================
+
+.. image:: https://img.shields.io/github/workflow/status/twisted/twisted-iocpsupport/Build%20and%20upload%20to%20PyPI?label=GitHub%20Actions&logo=github
+    :alt: GitHub Actions
+    :target: https://github.com/twisted/twisted-iocpsupport/
```
Comment (adiroiban) [inline README.rst:6]: Why ldaptor? 😀

File: README.rst [hunk 3]
```diff
@@ -1 +1,25 @@
...
+.. image:: https://img.shields.io/pypi/v/twisted-iocpsupport?logo=pypi
+    :alt: PyPI
+    :target: https://pypi.org/project/twisted-iocpsupport/
+
+
+An extension for use in the L{twisted.internet.iocpreactor} I/O Completion
+Ports reactor.
+
+This code was initially part of the core Twisted project. It was moved into a
```
Comment (adiroiban) [inline README.rst:15]: Maybe start with a brief summary and then explain all the details.

```
**This is a protected package. It should not be used outside of Twisted project.**
```

**Review by adiroiban — COMMENTED** [review]
I am happy with the current PR.

I will let @exarkun to do the final approval.

Thanks!

File: README.rst (hunk 2 above)
Comment (graingert) [reply README.rst:6]: An accidental, I pushed a sneaky fix though

File: README.rst (hunk 3 above)
Comment (graingert) [reply README.rst:15]:
```suggestion
**This is a protected package. It should not be used outside of Twisted project.**
This code was initially part of the core Twisted project. It was moved into a
```

Comment (graingert) [reply README.rst:15]: hmm I think what we have is clear enough for a v1.0.0, we can always do a v1.0.0.post0 for readme changes

**Review by adiroiban — APPROVED** [review]
I have checked with @exarkun over IRC and he will not have time to review this soon.

To not block this, I think that we should merge and release it.

Thanks

**Comment by adiroiban** [comment]
Should we name the default branch "default" or "trunk" to match twisted convention?

**Comment by altendky** [comment]
Given that `trunk` is really an SVN convention (right?), perhaps the move away from `master` in git is a reasonable time to bring twisted inline?  I assume this won't happen.

**Pull request merged**